spec/requests/application_controller_spec.rb

```diff
@@ -143,6 +143,21 @@
         expect(response.status).to eq(404)
         expect(response.body).to_not include('google.com/search')
       end
+
+      it 'should cache results' do
+        $redis.set('not_found_page_topics', 'first response')
```

Comment:
I personally think that we're not sufficiently testing the changes here. How I will go about testing this change will be as follows:

1. Create a topic and test that a not found will show that topic. 
1. Assert that the right key has been set in Redis
1. Create a new topic and test the same route.
1. Assert that the response does not include the new topic because the previous response should have been cached.